backend/tests/integration/orcid/test_orcid_domain_models.py

```diff
@@ -467,21 +468,39 @@ def test_happy_flow_put(self):
             self.factory.record_metadata.json["dois"] = dois
             self.factory_clashing.record_metadata.json["deleted"] = True
 
-            pusher = domain_models.OrcidPusher(self.orcid, self.recid, self.oauth_token)
-            result_putcode = pusher.push()
-
-        assert self.putcode == result_putcode
-        assert not self.cache.has_work_content_changed(self.inspire_record)
+            with mock.patch(
+                "inspirehep.orcid.tasks.orcid_push.retry"
+            ) as celery_task_with_retry:
+                celery_task_with_retry.side_effect = (
+                    Retry(exc=exceptions.DuplicatedExternalIdentifierPusherException),
+                )
```

Comment:
why is this needed? because of eager mode?

Reply:
Yes, it's same as here https://github.com/inspirehep/inspirehep/blob/642836ac47b99bab9bc578b382c10ca6892ceae2/backend/tests/integration/orcid/test_orcid_domain_models.py#L448-L460

Reply:
it's not the same. There you want to simulate exceeding the time limit. There's no such simulation happening here, this should be the standard behavior of the celery task. Did you try without the mocking?